ruby/ext/google/protobuf_c/extconf.rb

```diff
@@ -2,7 +2,9 @@
 
 require 'mkmf'
 
-$CFLAGS += " -O3 -std=c99 -Wno-unused-function -DNDEBUG "
+$CFLAGS += " -O3 -std=c99 -Wno-unused-function " +
+           "-Wno-declaration-after-statement -Wno-unused-variable " +
```

Comment:
I am... somewhat perplexed why this is. We're clearly passing in the `-std=c99` flag, and that `$CFLAGS` clearly takes effect because adding the warning suppressions cleans up the output. Perhaps `mkmf` knows better and appends some other flags? I'll try to dig into this a bit more.